### DISCUSSION TITLE
Unable to pass custom json body using docker image of perf test but works with source code as JAR

### DISCUSSION THREAD

**Discussion by bipul-jha** [rabbitmq/rabbitmq-perf-test#384] [General]
Hello perf-test team,

I was trying to use this tool to do performance test for our queues, when I tried to use `docker run -it --rm --network perf-test pivotalrabbitmq/perf-test:latest --uri amqp://rabbitmq --body <some_json_file.json>` it throws the same error all the time. 

"java.lang.IllegalArgumentException: some_json_file.json isn't a valid body file."

But when we tried to clone the repo and run with the JAR, it works! 

Used the below command for the same : 
`java -jar perf-test.jar --uri amqp://rabbitmq --body <some_json_file.json>`

Not sure whats the issue here. Can you look into this issue?

**Comment by michaelklishin** [comment]
`java -jar` loads the file from the local filesystem. `docker run -it` loads it from the container filesystem. If you don't mount any local files or directories to the container FS, Docker won't be able to locate and access the file.

**Reply by bipul-jha** [reply]
Thanks much @michaelklishin. Also we wanted to ask regarding sending payload (--body), it takes the files as comma separated values. So if we want to hit lets say 200 files/sec, we need to pass 200 json files as configuration ?

**Reply by acogoluegnes** [reply]
PerfTest will use the specified files for the body of messages. If you specify 1 file, all messages will have the same body, if you specify 5 files, messages will have 5 different bodies, etc. The typical use case for this feature is to have PerfTest running only publishers to fill a queue with real messages and have real application consumers processing these messages. So you can test real-life message processing.

You can specify 10 files and use the `--rate` option to specify the publishing rate.

**Reply by acogoluegnes** [reply]
Note there's #224 open to make the same features more flexible (making the files templates instead of fixed files).

**Reply by bipul-jha** [reply]
Thanks @acogoluegnes for the quick response. 
One more question, giving `-u` as a configuration creates new bindings in the existing queue during Perf-test simulation runs. Can we test the existing queue with no additional bindings ?

**Reply by acogoluegnes** [reply]
You can use the `--predeclared` flag and PerfTest will not create anything.

**Reply by bipul-jha** [reply]
Thanks much! We might need to update the documentation for this as `--exchange` and `--type` wasn't mentioned anywhere, I figured out from the command line's --help section. Previously I was of the opinion that we could only put load in queues and not exchange.
